src/test/java/hudson/tasks/junit/CaseResultTest.java

```diff
@@ -222,42 +225,38 @@ public void testAge() throws Exception {
         //Create a job:
         FreeStyleProject p = rule.createFreeStyleProject(projectName);
         p.getPublishersList().add(new JUnitResultArchiver("*.xml"));
-
-        //First build execution:
+        // add builder copying test result file
         p.getBuildersList().add(new TestBuilder() {
             public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-                build.getWorkspace().child("junit.xml").copyFrom(
-                    getClass().getResource(testResultResourceFile));
+                FilePath junitFile = build.getWorkspace().child("junit.xml");
+                junitFile.copyFrom(getClass().getResource(testResultResourceFile));
+                // sadly this can be flaky for 1ms.... (but hey changing to nano even in core might complicated :))
+                junitFile.touch(System.currentTimeMillis()+1L);
```

Comment:
So if `parseOldReports` now defaults to true, all the complicated logic to test this feature can be isolated to a handful of test cases, leaving the rest of the test code free of flaky timestamp manipulation, right?